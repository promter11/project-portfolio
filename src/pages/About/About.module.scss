@import "../../styles/variables";

.about {
  height: 100%;
  padding: 8rem 14rem;
  background: rgba(0, 0, 0, 0.75) url("../../assets/images/About.jpg") no-repeat
    center center / cover;
  background-blend-mode: darken;
}

.title {
  margin-bottom: 8rem;
  text-align: right;
  letter-spacing: 1rem;
  opacity: 0.25;
}

.block {
  position: relative;
  max-width: 67rem;
  margin: 0 auto;
  padding: 6rem 8rem;
  border-radius: 0.4rem;
  box-shadow: 0 0 1rem 0.5rem rgba(0, 0, 0, 0.1);
  background-color: $about-block-color;
  color: $black-color;
}

.subtitle,
.text {
  margin-bottom: 4rem;
  font-size: 1.8rem;
}

.subtitle {
  display: block;
  font-weight: 900;
  line-height: 2.4rem;
  text-transform: capitalize;
}

.text {
  font-weight: 100;
  line-height: 200%;
  text-align: justify;
}

.link {
  display: inline-block;
  flex-shrink: 0;
  padding: 1.5rem 3rem;
  font-size: 1.4rem;
  font-weight: 100;
  line-height: 100%;
  text-transform: uppercase;
  letter-spacing: 0.02rem;
  border: 0.1rem solid $black-color;
  transition: all 0.25s ease-in-out;
  background-color: transparent;
  color: $black-color;

  &:hover {
    background-color: $black-color;
    color: $white-color;
  }

  &:active {
    transition: none;
    transform: translateY(0.2rem);
  }
}
