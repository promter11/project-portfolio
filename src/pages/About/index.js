import React from "react";

import Heading from "../../components/Heading";
import Container from "../../components/Container";

import { aboutText } from "../../common/jsonText/aboutText";

import Resume from "../../assets/docs/Resume.pdf";
import styles from "./About.module.scss";

const About = ({ language }) => {
  return (
    <section className={styles.about}>
      <Container>
        <Heading className={styles.title} level={1}>
          {aboutText[language].title}
        </Heading>
        <div className={styles.block}>
          <Heading className={styles.subtitle} level={3}>
            {aboutText[language].subtitle}
          </Heading>
          {aboutText[language].text.map((item, index) => {
            return (
              <p key={index} className={styles.text}>
                {item}
              </p>
            );
          })}
          <a className={styles.link} href={Resume} download>
            {aboutText[language].link.text}
          </a>
        </div>
      </Container>
    </section>
  );
};

export default About;
