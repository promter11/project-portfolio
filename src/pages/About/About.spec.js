import React from "react";
import { shallow } from "enzyme";

import About from "./index";

const props = {
  language: "RU",
};

describe("About page", () => {
  it("renders correctly with props", () => {
    const wrapper = shallow(<About {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
