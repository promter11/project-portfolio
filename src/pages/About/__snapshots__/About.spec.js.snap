// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page renders correctly with props 1`] = `
<section
  className="about"
>
  <Container>
    <Heading
      className="title"
      level={1}
    >
      Обо мне
    </Heading>
    <div
      className="block"
    >
      <Heading
        className="subtitle"
        level={3}
      >
        Несколько слов обо мне...
      </Heading>
      <p
        className="text"
        key="0"
      >
        Я фронтенд-разработчик, занимающийся созданием пользовательских интерфейсов с использованием React.
      </p>
      <p
        className="text"
        key="1"
      >
        Я хорошо организован, мотивирован, ориентирован на работу в команде и постоянно работаю над улучшением своих профессиональных навыков. Уделяю внимание деталям, но без излишнего перфекционизма, когда это наносит ущерб скорости разработки.
      </p>
      <p
        className="text"
        key="2"
      >
        Заинтересован в работе над амбициозными проектами на актуальном стеке технологий с коллегами, имеющими большой опыт во front-end разработке, которым нравится делиться знаниями.
      </p>
      <a
        className="link"
        download={true}
        href="../files/resume.pdf"
      >
        Скачать резюме
      </a>
    </div>
  </Container>
</section>
`;
